### PR TITLE
gropp2020implicit

### PR DESCRIPTION
**Pull request by samatharhay**

### CODE REVIEW

File: _summaries/gropp2020implicit_2.md [hunk 1]
```diff
@@ -1,19 +1,85 @@
...
-score: # How did you like this paper 0(dislike) to 10(love)
+paper: gropp2020implicit
+author: # samatharhay
+score: # 7
 ---
 
-TODO: Summarize the paper:
 * What is the core idea?
+
+The core idea of the paper was to introduce a new paradigm built on implicit geometric regularization for computing high fidelity implicit neural representations directly from raw data scans.
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:11]: Might be good to talk about Eikonal equations here (top-left of page 3) since this is how this paper differs from the others. Functions that solve Eikonal equations automatically have unit normals (gradients).

File: _summaries/gropp2020implicit_2.md [hunk 2]
```diff
@@ -1,19 +1,85 @@
...
+
 * How is it realized (technically)?
+
+Given an input cloud X = {x_i}iEI C R^3 with or without normal data, N = {n_i} iEI C R^3, the goal is to compute the parameters \theta of a multi-layer perceptron (MLP) f(x;\theta), where f has the shape R^3 x R^m \rightarrow R, such that it approximates a signed distance function to a plausible surface M defined by X and N.
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:15]: Use $$s for LaTeX, but I think you already know.

File: _summaries/gropp2020implicit_2.md [hunk 3]
```diff
@@ -1,19 +1,85 @@
...
+
+To help learn these parameters the authors propose the loss function:
+
+![image of function 2](gropp2020implicit_2a.png)
+
+where \lambda > 0 is a hyperparameter, ||.|| = ||.||2 (the euclidean 2-norm) and
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:21]: Use $$\\| . \\|$$ for the bars, otherwise it'll look like this:

image

File: _summaries/gropp2020implicit_2.md [hunk 4]
```diff
@@ -1,19 +1,85 @@
...
+
+where \lambda > 0 is a hyperparameter, ||.|| = ||.||2 (the euclidean 2-norm) and 
+
+![image of function 3](gropp2020implicit_2b.png)
+
+which encourages f to vanish on X.
+
+The second term in the first equation is called the Eikonal term, which encourages the gradients to be of unit 2-norm.
+
+Optimizing the loss function using SGD, or a variant of SGD) results in solutions that are close to a signed distance function with a smooth and suprising ly plausible zero level set.
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:29]: surprising ly --> surprisingly

Consider mentioning the Implicit Geometrical Regularization property here as a consequence of the plane reproduction property. Also that the plane reproduction property means the loss function will always arrive at a minima when optimizing with SGD or variants.

Then tell the readers to read the proof of plane reproduction in the paper if interested.

File: _summaries/gropp2020implicit_2.md [hunk 5]
```diff
@@ -1,19 +1,85 @@
...
+Regular Grid Volumetric Function (RGVF):
+* implicit function is defined only at grid points, requiring an interpolation scheme to extend it to interior of cells
+* It requires cubic-size grid and is not data-dependent meaning it does not adhere to the specific geometry of the shapes one wishes to approximate
+* a version of the Eikonal regularization term has been used as a normalization term, as apposed to this work which uses it alone as regularizing the zero level set.
+* gradients are computed using finite differences or via some fixed basis function.
+
+Neural parametric surfaces:
+* finding a consistent atlas covering of the surface can be challenging.
+* the shape can be easily sampled.
+* challenging to produce a set of perfectly overlapping charts, a property that holds by construction for implicit representations.
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:48]: I don't think the related works are necessary, but you can mention the parts you think are important that this paper differs from others.

File: _summaries/gropp2020implicit_2.md [hunk 6]
```diff
@@ -1,19 +1,85 @@
...
 * How well does the paper perform?
+
+To test performance a 8 layer MLP, with hidden size 512 and a single skip connection from the inpu to the middle layer, was used. 
+They used geometric initializaiton and \lambda = 0.1 and \tau = 1.
+They defined the distribution D as the average of a uniform distribution and a sum of Gaussians centered at X with standard debiation equal to the distance to the kth nearest neighbor (k =50).
+For evaluation they use Chamfer and Hausdorff distances:
+
+![equations 10-13](gropp2020implicit_2e.png)
+
+They first evaluated the model by testing its ability to reproduce a signed distance function to known maifold surfaces
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:60]: maifold --> manifold

File: _summaries/gropp2020implicit_2.md [hunk 7]
```diff
@@ -1,19 +1,85 @@
...
+ ![insert figure 5](gropp2020implicit_2g.png)
+
+They then conducted experiments on the task of surface reconstruction.
+They evaluated their method on the surface reconstruction benchmark from Berg elt al., 2013 using input data from william et al. 2019b.
+The data consisted of 5 shapes with non trivial tropology or details.
+
+ ![table 2](gropp2020implicit_2h.png)
+
+Lastly, they experimented on learning shape space. 
+They used a datset of high resolution raw scans of 10 humans in multiple poses (8 humnas for training and 2 for testing).
```
Comment (liaojh1998) [inline _summaries/gropp2020implicit_2.md:75]: datset --> dataset

**Review by liaojh1998 — CHANGES_REQUESTED** [review]
This is a good summary. I think it's good to just copy the entire first part of the method section for the loss function. The results are summarized nicely. The formatting needs some change. It would be good to mention the loss optimizes the function to solve Eikonal equations though.

@samatharhay @mmcinnestaylor I'm not particularly well-versed in implicit neural representations either, but I think y'all get the idea that it's trying to train a neural network to learn the shape of a surface given its 3D point cloud. I strongly recommend this [11-minute presentation](https://www.youtube.com/watch?v=6cOvBGBQF9g&ab_channel=AmosGropp) from the authors of this paper for a nice high-level overview. I can also give an overview of what I think the paper is trying to say in sections:

1. An introduction to implicit neural representations.
2. The loss for optimizing a solution f (the neural network) to the Eikonal equations. This loss used 1) L1 regression on f(x), 2) L2 regression on gradients of f(x) to = the normals of x, and 3) some interesting expectation term called Eikonal term, but it's found to encourage the L2 norm of gradients of f(x) to = 1? This loss is always found to converge to a minimum (the signed distance function we want for approximating the shape) by SGD. Also, if SGD chooses a solution out of the infinite number of solutions, it will choose a good one that looks smooth (as a consequence of the plane reproduction property). They call this property "Implicit geometrical regularization".
3. Previous works, probably don't matter as much for the class presentation.
4. The proof of the plane reproduction property. That is, SGD on the loss will always converge to a global minimum under some constraints (linear network case). They claim that 1) GD will avoid strict saddle points, and 2) GD will not go to a maxima because the loss can be positive infinity, so SGD must converge to a minima because in general, "MLP is approximately linear".
5. The MLP architecture and how it's evaluated. They take multiple points, x, in some 3D space, and see if f(x) = 0. The ones that equate are the points on the surfaces. Then, the Chamfer and Hausdorff distances try to match the closest ground truth points to these points on the surfaces and see how far apart they are.
6. Evaluation on toy manifolds (plane, sphere, and Bimba), which looks great. Then, evaluated on D-Faust between IGR (this paper's model) and a regression model.
7. Beat DGP on surface reconstruction dataset on 4 out of 5 experiments. Analysis of D-Faust experiments (shape space, multiple shapes, shape space exploration).
8. Conclusions.

**Review by liaojh1998 — APPROVED** [review]
Looks good. The formatting is done well now and the statement on plane reproduction is nice. Thanks for working on this again.
